stock_storage_type/models/stock_location.py

```diff
@@ -314,9 +327,12 @@ def _compute_location_is_empty(self):
             # to enforce concurrent transaction safety: 2 moves taking
             # quantities in a location have to be executed sequentially
             # or the location could remain "not empty"
+            location_domain = [("location_id", "=", rec.id)]
+            domain = OUT_MOVE_LINE_DOMAIN + location_domain
+            out_move_lines = self.env["stock.move.line"].search(domain, order="id")
+            quants = self.env["stock.quant"].search(location_domain, order="id")
```

Comment:
As discussed let's move these 2 queries out of the look, in case `location_is_empty` is computed for may locations at once.